gslab_scons/misc.py

```diff
@@ -264,29 +161,47 @@ def get_directory(path):
     directory = os.path.dirname(path)
     if directory == '':
         directory = './'
-
     return directory
 
 
-def check_targets(target_files):
+def get_executable(language_name, manual_executables = {}):
     '''
-    This function raises an exception if any of the files listed as `target_files`
-    do not exist after running a builder. 
+    Get executable stored at language_name of dictionary manual_executables.
+    If key doesn't exist, use a default.
     '''
-    if not isinstance(target_files, list):
-        target_files = [target_files]
-    non_existence = []
-    for target in target_files:
-        target = str(target)
-        if not os.path.isfile(target):
-            non_existence.append(target)
-
-    if non_existence:
-        error_message = 'The following target files do not exist after build:\n' + \
-            '\n'.join(non_existence)
-        raise _exception_classes.TargetNonexistenceError(error_message)
-
-    return None
+    default_executables = {
+        'python': 'python',
+        'r': 'Rscript',
+        'stata': '',
+        'matlab': 'matlab',
+        'lyx': 'lyx',
+        'latex': 'pdflatex',
+        'tablefill': '',
+        'anything builder': ''
+    }
+    lower_name = language_name.lower().strip()
+    manual_executables = {str(k).lower().strip(): str(v).lower().strip() 
+                          for k, v in manual_executables.items()}
+    manual_executables = {k: v for k, v in manual_executables.items() 
+                          if k and v and v not in ['none', 'no', 'false', 'n', 'f']}
+    try:
+        executable = manual_executables[lower_name]
+    except KeyError:
+        has_default_executable = lower_name in default_executables.keys()
+        if not has_default_executable:
+            error_message = 'Cannot find default executable for language: %s. ' \
+                            'Try specifying a default.' % language_name
+            raise _exception_classes.PrerequisiteError(error_message)
+        elif lower_name != 'stata':
+            executable = default_executables[lower_name]
+        elif is_unix():

```

Comment:
I went with your logic. Writing it with dictionaries felt weird and contrived because I had to hard-code the key names in two locations. If we end up with a lot of os-dependent executables down the road, then we can work on generalizing it.